### PR TITLE
Support passing array of metadata value

### PR DESCRIPTION
**Pull request by jiexinhuang**
It is quite common that we want to record multiple values for a metadata key when a money transfer is performed. For example: a money transfer can have multiple tags, categories, applied taxes.

This change will allow DoubleEntry gem users to do this:
```
DoubleEntry.transfer(
  amount,
  :from  => one_account,
  :to      => another_account,
  :code => :some_money_movement,
  :metadata => {:key1 => ['value 1', 'value 2'] }
```

Also added some test to ensure reporting still works correctly for money transfers like this.

### CODE REVIEW

File: lib/double_entry/transfer.rb [hunk 1]
```diff
@@ -115,8 +115,15 @@ def create_lines(amount, code, detail, from_account, to_account)
...
-        LineMetadata.create!(:line => debit, :key => key, :value => value)
+        if value.is_a?(Array)
+          value.each do |each_value|
+            LineMetadata.create!(:line => credit, :key => key, :value => each_value)
+            LineMetadata.create!(:line => debit, :key => key, :value => each_value)
+          end
+        else
+          LineMetadata.create!(:line => credit, :key => key, :value => value)
+          LineMetadata.create!(:line => debit, :key => key, :value => value)
+        end
```
Comment (spickermann) [inline lib/double_entry/transfer.rb]: Did our consider using `Array()` instead of the `is_a?(Array)`. Because `Array` ensures that there is always an array it would supersede the  `if ... else` condition.

```ruby
Array(value).each do |each_value|
  LineMetadata.create!(:line => credit, :key => key, :value => each_value)
  LineMetadata.create!(:line => debit, :key => key, :value => each_value)
end
```

Comment (jiexinhuang) [inline lib/double_entry/transfer.rb]: Heya, @spickermann 
TIL, that is a great trick. I didn't know Ruby allows this coercion.

File: spec/double_entry/reporting_spec.rb [hunk 2]
```diff
@@ -201,6 +201,34 @@
         end
       end
 
+      context 'filtering by key with multiple values' do
+        let(:code) { :spend }
+        let(:filter) do
+          [
+            :metadata => {
+              :category => category,
```
Comment (spickermann) [inline spec/double_entry/reporting_spec.rb]: I wonder if it makes sense to allow a list of categories in the filter? But we would have to make the decision if an entry had to match *all categories* or just *any category* in that list. Perhaps this is out of scope at the moment...

Comment (jiexinhuang) [inline spec/double_entry/reporting_spec.rb]: It will be a useful feature, but as you mentioned, there is the decision around matching all categories or any category.
Another PR welcome! :trollface:

File: lib/double_entry/transfer.rb (hunk 1 above)
Comment (stephnacios) [inline lib/double_entry/transfer.rb]: TIL!  Neat!

**Comment by stephnacios** [comment]
Looks great! Nice work :100:

File: spec/double_entry/reporting_spec.rb (hunk 2 above)
Comment (orien) [inline spec/double_entry/reporting_spec.rb]: Hi @spickermann! That indeed would be a useful feature. I don't know how to resolve the and/or problem at this point. We don't yet have a use case for it here at Envato so I'm going to play the 'out of scope' card this time. 😀

**Comment by orien** [comment]
+1
